### PR TITLE
feat: [PIE-5178]: add tests for services and bug fixes

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vkbansal-harness** [comment]
retrigger titlecheck

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 11 Issues 
* Code issue 11 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11212)

File: src/modules/10-common/components/MultiTypeMulliSelectDropDown/MultiTypeMultiSelectDropDown.module.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.main {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/MultiTypeMulliSelectDropDown/MultiTypeMultiSelectDropDown.module.scss:1]: nit: the folder name contains Mulli 🙈

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx [hunk 2]
```diff
@@ -501,20 +524,27 @@ export default function DeployServiceEntityWidget({
                     />
                   ) : null}
                 </Layout.Horizontal>
+                {isMultiSvc ? (
+                  <FormInput.CheckBox
+                    label={getString('cd.pipelineSteps.serviceTab.multiServicesParalleDeployLabel')}
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:529]:
```suggestion
                    label={getString('cd.pipelineSteps.serviceTab.multiServicesParallelDeployLabel')}
```

File: src/modules/10-common/components/MultiTypeMulliSelectDropDown/MultiTypeMultiSelectDropDown.module.scss (hunk 1 above)
Comment (vkbansal-harness) [reply src/modules/10-common/components/MultiTypeMulliSelectDropDown/MultiTypeMultiSelectDropDown.module.scss]: Fixed

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    await waitFor(() => expect(container.querySelector('.bp3-spinner')).not.toBeInTheDocument())
+
+    expect(container).toMatchSnapshot()
+
+    const toggle = container.querySelector('.Toggle--input')!
+
+    act(() => {
+      userEvent.click(toggle)
+    })
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx:59]: act wrap can be removed

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    userEvent.click(multi)
+
+    await waitFor(() => expect(document.body.querySelector('.bp3-menu')).toBeInTheDocument())
+
+    const menu = document.body.querySelector<HTMLDivElement>('.bp3-menu')!
+
+    await act(async () => {
+      const svc1 = await findByTextGlobal(menu, 'Service 1')
+      userEvent.click(svc1)
+    })
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx:72]: `act` can be removed here, the below assertion can be wrapped with `waitFor`

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.SingleService.test.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    expect(container).toMatchSnapshot()
+
+    userEvent.click(container.querySelector('[data-icon="chevron-down"]')!)
+
+    await waitFor(() => expect(document.body.querySelector('.bp3-menu')).toBeInTheDocument())
+
+    const menu = document.body.querySelector<HTMLDivElement>('.bp3-menu')!
+
+    await act(async () => {
+      const svc1 = await findByTextGlobal(menu, 'Service 1')
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.SingleService.test.tsx:59]: `const svc1 = await findByTextGlobal(menu, 'Service 1')` doesn't have to be nested inside act

File: src/modules/75-cd/strings/strings.en.yaml [hunk 6]
```diff
@@ -186,6 +186,7 @@ pipelineSteps:
     newServiceOverride: New configuration overrides
     multiServicesText: Deploy to multiple Services
     multiServicesTitleText: Deploy to multiple services?
+    multiServicesParalleDeployLabel: Deploy services in parallel?
```
Comment (rameshx) [inline src/modules/75-cd/strings/strings.en.yaml:189]:
```suggestion
    multiServicesParallelDeployLabel: Deploy services in parallel?
```

**Review by rameshx — COMMENTED** [review]
All `act`s wrapping `userEvent` can be removed.

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx (hunk 3 above)
Comment (vkbansal-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.MultiService.test.tsx:59]: If I remove this, I keep getting warnings. not wrapped in act. hence added this.

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx (hunk 2 above)
Comment (udayvunnam) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:529]: +1

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityUtils.ts [hunk 7]
```diff
@@ -55,23 +57,28 @@ export function getValidationSchema(getString: UseStringsReturn['getString']): Y
...
               value.services.length > 0 &&
               value.services.every(row => typeof row.value === 'string' && row.value.length > 0)
 
+            /* istanbul ignore else */
             if (hasValues) {
               return true
             }
           }
 
+          /* istanbul ignore else */
```
Comment (udayvunnam) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityUtils.ts:81]: Are all these checks for the type narrowing only?

If not we should be able to test cover/remove comments
If yes, we can have some standard utility that's ignored by tests by default, We can explore something like https://www.npmjs.com/package/tiny-invariant

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.SingleService.test.tsx [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+
+    expect(dialog).toMatchSnapshot()
+
+    const name = queryByAttribute('name', dialog, 'name')!
+
+    userEvent.type(name, 'Service 4')
+
+    const save = await findByTextGlobal(dialog, 'save')
+
+    await waitFor(() => expect(save.parentElement!.classList).not.toContain('bp3-disabled'))
```
Comment (udayvunnam) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/__tests__/DeployServiceEntityWidget.SingleService.test.tsx:119]: not: doesn't toBeDisabled work here ?

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityUtils.ts (hunk 7 above)
Comment (vkbansal-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityUtils.ts:81]: Yes these are just for type narrowing. Invariant will throw. We do not want to throw.

**Review by YounixIO — APPROVED** [review]

**Comment by vkbansal-harness** [comment]
trigger cypress

**Pull request merged**